### PR TITLE
Gutenberg: alternative take at Calypsoify iframe

### PR DESCRIPTION
**Pull request by vindl**
#### Changes proposed in this Pull Request

* Alternative take at https://github.com/Automattic/wp-calypso/pull/30307 which utilizes feature flags. If needed this will allow us for easier distribution of tasks and testing. To make this easier to grok, I tried to keep the amount of changes to a minimum, and avoided refactoring of existing flags for now.

#### Testing instructions

- Same as https://github.com/Automattic/wp-calypso/pull/30307

### CODE REVIEW

**Comment by kwight** [comment]
Works wonderfully (it's not even that slow 😱 ). I noticed the green "View Post" banner link (after publishing) needs to be bumped up to the parent frame. I do like how much clearer this PR is compared to   #30307 .

**Comment by Stojdza** [comment]
@vindl @gwwar I added `Needs e2e Testing` label, and 33 tests failed with the same error - `Could not locate the editor iFrame.`

File: config/wpcalypso.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
 		"blogger-plan": false,
 		"comments/management/threaded-view": false,
 		"calypsoify/gutenberg": true,
+		"calypsoify/iframe": false,
```
Comment (gwwar) [inline config/wpcalypso.json]: So for the E2E tests to see the iframe, we'll need to flip this on I think. cc @Automattic/flowpatrol 

I'll add a commit in a bit

**Comment by gwwar** [comment]
@Automattic/flowpatrol after adding the flag for wpcalypso, it does look like we have a clean run on the canaries. Let us know if you need help with any remaining suites. I'd like to land this once we know that we don't break full E2Es on master.

File: config/wpcalypso.json (hunk 1 above)
Comment (gwwar) [reply config/wpcalypso.json]: Added in 9cca290

**Comment by Stojdza** [comment]
@gwwar Now we have clean run of e2e tests. I made some changes, but I've also disabled several tests because of the issues:

- One is already addressed [here](https://github.com/Automattic/wp-calypso/pull/30307#issuecomment-458365266) 

- The other one is also related to redirection, but the page is broken. This page occurs after user clicks on "Move to Trash":
screen shot 2019-01-31 at 12 32 51



As I mentioned, I've disabled some tests in[ this PR ](https://github.com/Automattic/wp-e2e-tests/pull/1784) if this needs to land ASAP. If not, we can wait and enable this tests once these issues are solved.

**Comment by gwwar** [comment]
@Automattic/flowpatrol cc @Stojdza @bsessions85  Let's not land https://github.com/Automattic/wp-e2e-tests/pull/1784. Keep the diff, then update the branch to match our next PR.

We'll drop the wpcalypso flag and keep that false for this feature. In the next follow up PR we'll enable it again so folks can continue with E2E test updates. We'll continue to drop this until we're further along in functionality.

**Comment by gwwar** [comment]
Note that CI tests will be red due to flags / a WIP test branch https://github.com/Automattic/wp-e2e-tests/pull/1784

I'll try landing this portion and will revert if staging canaries / master tests do break.

**Pull request merged**